components/builder-db/src/models/channel.rs

```diff
@@ -382,12 +387,101 @@ impl From<JobGroupTrigger> for PackageChannelTrigger {
+}
+
+#[derive(Debug, Serialize, Deserialize, Queryable, PartialEq)]
+pub struct AuditPackage {
+    pub package_ident:  BuilderPackageIdent,
+    pub channel:        String,
+    pub operation:      PackageChannelOperation,
+    pub trigger:        PackageChannelTrigger,
+    #[serde(with = "db_id_format")]
+    pub requester_id:   i64,
+    pub requester_name: String,
+    pub created_at:     Option<NaiveDateTime>,
+    pub origin:         String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct AuditPackageEvent {
+    pub operation:     PackageChannelOperation,
+    pub created_at:    Option<NaiveDateTime>,
+    pub origin:        String,
+    pub channel:       String,
+    pub package_ident: BuilderPackageIdent,
+}
+
+impl AuditPackage {
+    pub fn list(el: ListEvents, conn: &PgConnection) -> QueryResult<(Vec<AuditPackage>, i64)> {
+        Counter::DBCall.increment();
+        let start_time = Instant::now();
+
+        let mut query = audit_package::table.left_join(
+                origin_packages::table.on(origin_packages::ident.eq(audit_package::package_ident))
+            )
+            .select(audit_package::all_columns)
+            .distinct_on((audit_package::package_ident, audit_package::created_at))
+            .into_boxed();
+
+        if let Some(session_id) = el.account_id {
+            let origins = origin_members::table.select(origin_members::origin)
+                                               .filter(origin_members::account_id.eq(session_id));
+            query = query.filter(
+                origin_packages::visibility
+                    .eq(any(PackageVisibility::private()))
+                    .and(origin_packages::origin.eq_any(origins))
+                    .or(origin_packages::visibility.eq(PackageVisibility::Public))
+                    .or(audit_package::requester_id.eq(session_id)),
+            );
+        } else {
+            query = query.filter(origin_packages::visibility.eq(PackageVisibility::Public));
+        }
+
+        if el.last_n_days > 0 {
+            query =
+                query.filter(audit_package::created_at.gt((now.into_sql::<Timestamptz>()
+                                                           - el.last_n_days.days()).nullable()));
+        }
+
+        if !el.channel.is_empty() {
+            query = query.filter(audit_package::channel.eq(el.channel));
+        }
+
+        let query = query.order((audit_package::created_at.desc(),
+                                 audit_package::package_ident.desc()))
+                         .paginate(el.page)
+                         .per_page(el.limit);
+        let (events, total_count): (std::vec::Vec<AuditPackage>, i64) =
+            query.load_and_count_records(conn)?;
+        let duration_millis = start_time.elapsed().as_millis();
+        Histogram::DbCallTime.set(duration_millis as f64);
+
+        Ok((events, total_count))
+    }
+}
+
+impl Into<AuditPackageEvent> for AuditPackage {
```

Comment:
better to implement `From` on `AuditPackageEvent` since that gives you `into` for free and more recent versions of rust will warn on that.